tests/misc/test_split.py

```diff
@@ -145,27 +147,26 @@ class SimpleSplitTests(unittest.TestCase):
         'foo\nbar': ['foo', '\nbar'],
     }
 
-    def test_str_split(self):
+    @pytest.mark.parametrize('test', TESTS)
+    def test_str_split(self, test):
         """Test if the behavior matches str.split."""
-        for test in self.TESTS:
-            with self.subTest(string=test):
-                self.assertEqual(split.simple_split(test),
-                                 test.rstrip().split())
+        assert split.simple_split(test) == test.rstrip().split()
 
     def test_str_split_maxsplit_1(self):
         """Test if the behavior matches str.split with maxsplit=1."""
-        string = "foo bar baz"
-        self.assertEqual(split.simple_split(string, maxsplit=1),
-                         string.rstrip().split(maxsplit=1))
+        s = "foo bar baz"
+        actual = split.simple_split(s, maxsplit=1)
+        expected = s.rstrip().split(maxsplit=1)
+        assert actual == expected
 
     def test_str_split_maxsplit_0(self):
         """Test if the behavior matches str.split with maxsplit=0."""
-        string = "  foo bar baz  "
-        self.assertEqual(split.simple_split(string, maxsplit=0),
-                         string.rstrip().split(maxsplit=0))
+        s = "  foo bar baz  "
+        actual = split.simple_split(s, maxsplit=0)
+        expected = s.rstrip().split(maxsplit=0)
```

Comment:
If I'm not mistaken ``test_str_split_maxsplit_1`` and ``test_str_split_maxsplit_0`` are another great use case of ``parametrize`` (``s`` and ``maxsplit``).